sdk/eventgrid/azure-messaging-eventgrid-cloudnative-cloudevents/src/main/java/com/azure/messaging/eventgrid/cloudnative/cloudevents/EventGridCloudNativeEventPublisher.java

```diff
@@ -0,0 +1,154 @@
+     * @param syncClient a service client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     */
+    public static void sendEvents(EventGridPublisherClient<com.azure.core.models.CloudEvent> syncClient,
+        Iterable<CloudEvent> events) {
+        syncClient.sendEvents(toEventGridCloudEvents(events));
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain.
+     *
+     * @param asyncClient a service asynchronous client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @return a Mono that completes when the events are sent to the service.
+     */
+    public static Mono<Void> sendEventsAsync(
+        EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient, Iterable<CloudEvent> events) {
+        return asyncClient.sendEvents(toEventGridCloudEvents(events));
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain and gives the response issued by EventGrid.
+     *
+     * @param syncClient a service client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @param context the context to use along the pipeline.
+     */
+    public static void sendEventsWithResponse(EventGridPublisherClient<com.azure.core.models.CloudEvent> syncClient,
+        Iterable<CloudEvent> events, Context context) {
+        syncClient.sendEventsWithResponse(toEventGridCloudEvents(events), context);
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain and gives the response issued by EventGrid.
+     *
+     * @param asyncClient a service asynchronous client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @return the response from the EventGrid service.
+     */
+    public static Mono<Response<Void>> sendEventsWithResponseAsync(
+        EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient, Iterable<CloudEvent> events) {
+        return asyncClient.sendEventsWithResponse(toEventGridCloudEvents(events));
+    }
+
+    private static Iterable<com.azure.core.models.CloudEvent> toEventGridCloudEvents(Iterable<CloudEvent> events) {
+        List<com.azure.core.models.CloudEvent> cloudEvents = new ArrayList<>();
+        for (CloudEvent event : events) {
+            cloudEvents.add(toEventGridCloudEvent(event));
+        }
+        return cloudEvents;
+    }
+
+    private static com.azure.core.models.CloudEvent toEventGridCloudEvent(CloudEvent event) {
+        if (event == null) {
+            return null;
+        }
+        // Identify data format by data content type
+        CloudEventDataFormat dataFormat;
+        final String dataContentType = event.getDataContentType();
+        if ("application/json".equals(dataContentType)) {
```

Comment:
I thought you were using the instance type of the data, rather than the data content type?

Reply:
I would @mssfang speak about it, but he discovered it's not reliable as users can pass any implementation of `CloudEventData`.
 
@mssfang  it seems spec says that [when the format is not provided, it should  be JSON](https://github.com/cloudevents/spec/blob/06e15360590bfcd55589331cb256279215d4d2f1/spec.md#datacontenttype) 

> In other words: a JSON-format event with no datacontenttype is exactly equivalent to one with datacontenttype="application/json".

Reply:
After reviewing with There is a couple of concern by adding extra dependencies:
- When using the `instanceof` approach, it depends on the source code implementation. The way of implementation could change, the same problem as using the `content type` approach. But in java, we need to import a type [JsonCloudEventData](https://github.com/cloudevents/sdk-java/blob/a94bc5c81c83d21b5d6ec367830b66b0d9569a6d/formats/json-jackson/src/main/java/io/cloudevents/jackson/JsonCloudEventData.java#L18), which is from another package, `io.cloudevents.jackson`, so we need to add the extra dependency, also, the jackson version they use is older than what we have in azure core. After talking with @lmolkova. We think it is straight to check the data content type.

Reply:
In that case, should we match the content type logic that is used in the CNCF library? To me, this seems even more risky as we are depending on implementation details, but as it stands right now, I don't think we are accurately assigning the format. In the CNCF version, "text/json" is also treated as a JSON string.

https://github.com/cloudevents/sdk-java/blob/a94bc5c81c83d21b5d6ec367830b66b0d9569a6d/formats/json-jackson/src/main/java/io/cloudevents/jackson/JsonFormat.java#L148

Reply:
@JoshLove-msft I'm not an expert here, but a couple of observations:
`datacontenttype` is actually [a standard](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#datacontenttype)  while `JsonFormat` is an implementation detail. I.e. I can use native CloudEvents Java SDK and populate json using any other library (not `cloudevents-json-jackson`) and have mine `OtherJsonFormat`, relying on any specific type here is not reliable.

Perhaps we need to dig into what are all possible json formats  and how they are populated with `datacontenttype`

Reply:
ah, I see, so this seem to be the approach cncf library takes `contentType == null || contentType.startsWith("application/json") || contentType.startsWith("text/json")`, which makes sense instead of just  `application/json"` we have now

Reply:
Yes, I think we will need to be consistent with the approach CNCF uses and hope that they don't change it.

Reply:
The other option, which is what we did in .NET, was to use the CNCF library to encode the CloudEvent into bytes, and then just send that directly to Event Grid.

Reply:
it's possible in Java, but requires us to write a custom serializer for CNCF CloudEvent (which is possible, but brings similar concerns to what we have with this kinda-copy code). Otherwise, we have to use another serialization library, which adds a new dependency (and it's not a good option)

Reply:
Ok, so actually gave more clarity (it's in master, not in 1.0.1 branch)

https://github.com/cloudevents/spec/blob/master/cloudevents/formats/json-format.md#311-payload-serialization

> That is, a datacontenttype declares JSON-formatted content if its media type, when stripped of parameters, has the form */json or */*+json. If the datacontenttype is unspecified, processing SHOULD proceed as if the datacontenttype had been specified explicitly as application/json

So `*/json`, `*/*+json` and `null` would cover 1.0.1 and vNext and seem to be a reliable solution. Thoughts?

Reply:
That's great that they are standardizing that. In C#, it treats anything starting with "text/" as JSON - https://github.com/Azure/azure-rest-api-specs/blob/9822cfaa38c926312e4e5208e7928dc588abaca2/specification/schemaregistry/data-plane/Microsoft.EventHub/stable/2021-10/schemaregistry.json.

If that is part of the spec, then it seems safe to rely on it.

Reply:
This still would cause a discrepancy for anyone using the released version of the Java CNCF library though. Should we match it for now, and then update to the official spec once the CNCF does? I'm not sure they will be able to adhere to this new standard without a new major version.

Reply:
sounds reasonable, especially since clarification comes in vNext and still subject to change